application/common/models/Mfa.php

```diff
@@ -55,20 +63,24 @@ public function fields(): array
                 }
                 return null;
             },
-            'data' => function ($model) {
-                $data = [];
-                /** @var Mfa $model */
-                if ($model->verified === 1 && $model->scenario === User::SCENARIO_AUTHENTICATE) {
-                    $data += $model->authInit();
-                }
-                if ($model->type === self::TYPE_BACKUPCODE || $model->type === self::TYPE_MANAGER) {
-                    $data += ['count' => count($model->mfaBackupcodes)];
-                }
-                return $data;
-            }
+            'data',
         ];
     }
 
+    /**
+     * @param string $rpOrigin The Relay Party Origin, used for WebAuthn and ignored for others
+     */
+    public function loadData(string $rpOrigin = '')
+    {
+        $this->data = [];
+        if ($this->verified === 1 && $this->scenario === User::SCENARIO_AUTHENTICATE) {
+            $this->data += $this->authInit($rpOrigin);
+        }
+        if ($this->type === self::TYPE_BACKUPCODE || $this->type === self::TYPE_MANAGER) {
+            $this->data += ['count' => count($this->mfaBackupcodes)];
+        }
```

Comment:
Why include the count of Backup Codes when the type is "manager"?

Reply:
I'm not sure, I left that as it was. I just moved it from the fields function to it's own.